lib/api/controllers/funnelController.js

```diff
@@ -374,6 +374,17 @@ class FunnelController {
       .then(responseData => {
         modifiedRequest.setResult(responseData, {status: request.status === 102 ? 200 : request.status});
 
+        if (!this.isNativeController(modifiedRequest) && !modifiedRequest.response.raw) {
+          // check if the plugin response can be serialized
+          try {
+            JSON.stringify(modifiedRequest.response.toJSON());
```

Comment:
This check should be performed directly on `responseData` to prevent the unnecessary overhead of `RequestResponse.toJSON`.

Reply:
:+1: